examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -138,4 +192,110 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
             findNavController().navigateUp()
         }
     }
+
+    private fun startPurchasePackage(
+        purchaseOption: PurchaseOption?,
+        currentPackage: Package,
+        upgradeInfo: UpgradeInfo?
+    ) {
+        if (upgradeInfo == null) {
+            if (purchaseOption == null) {
+                Purchases.sharedInstance.purchasePackageWith(
+                    requireActivity(),
+                    currentPackage,
+                    purchaseErrorCallback,
+                    successfulPurchaseCallback
+                )
+            } else {
+                Purchases.sharedInstance.purchaseSubscriptionOptionWith(
+                    requireActivity(),
+                    currentPackage.product,
+                    purchaseOption,
+                    purchaseErrorCallback,
+                    successfulPurchaseCallback
+                )
+            }
+        } else {
+            if (purchaseOption == null) {
+                Purchases.sharedInstance.purchasePackageWith(
+                    requireActivity(),
+                    currentPackage,
+                    upgradeInfo,
+                    purchaseErrorCallback,
+                    successfulUpgradeCallback
+                )
+            } else {
+                Purchases.sharedInstance.purchaseSubscriptionOptionWith(
+                    requireActivity(),
+                    currentPackage.product,
+                    purchaseOption,
+                    upgradeInfo,
+                    purchaseErrorCallback,
+                    successfulUpgradeCallback
+                )
+            }
+        }
+    }
+
+    private fun toggleLoadingIndicator(isLoading: Boolean) {
+        binding.purchaseProgress.visibility = if (isLoading) View.VISIBLE else View.GONE
+    }
+
+    private fun showOldSubIdPicker(callback: (String?) -> Unit) {
+        val activeSubIds = activeSubscriptions.map { it.split(":").first() } // TODOBC5 remove sub Id parsing
```

Comment:
we need this logic until `EntitlementInfo` and `CustomerInfo` split out sub and base plan ids.

Reply:
Is there a backend task for this in the backlog?

Reply:
[yep!](https://revenuecats.atlassian.net/browse/CF-1118)